### PR TITLE
fix(erc20): Generate salts using a nonce from disk

### PR DESCRIPTION
**Pull request by calbera**

### CODE REVIEW

File: cosmos/lib/salt.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	storetypes "cosmossdk.io/store/types"
+)
+
+const globalNonce = `globalNonce`
+
+var globalNonceKey = []byte(globalNonce)
+
+// UniqueDeterministicSalt returns a unique and deterministic salt for the given input bytes. Uses
+// sha256 to hash the input bytes and the global nonce and returns the salt as a *uint256.Int.
+func UniqueDeterminsticSalt(nonceStore storetypes.BasicKVStore, input []byte) *uint256.Int {
```
Comment (itsdevbear) [inline cosmos/lib/salt.go:39]: Ask ND he does a similar thing for TokenContainer.

Comment (itsdevbear) [reply cosmos/lib/salt.go:39]: don't need this whole lib thing.

File: cosmos/x/erc20/keeper/keeper.go [hunk 2]
```diff
@@ -47,15 +47,15 @@ func NewKeeper(
 	authority sdk.AccAddress,
 ) *Keeper {
 	return &Keeper{
-		storeKey:   storeKey,
+		StoreKey:   storeKey,
```
Comment (itsdevbear) [inline cosmos/x/erc20/keeper/keeper.go:50]: this should be private.

File: cosmos/lib/salt.go (hunk 1 above)
Comment (calbera) [reply cosmos/lib/salt.go:39]: thought its good practice to do it as a library so even ND can use this

File: cosmos/x/erc20/keeper/keeper.go (hunk 2 above)
Comment (calbera) [reply cosmos/x/erc20/keeper/keeper.go:50]: can make it private but have to regenerate everytime

File: cosmos/lib/salt.go (hunk 1 above)
Comment (itsdevbear) [reply cosmos/lib/salt.go:39]: yeah fair, though I don't see why we need to use a salt and CREATE2 when we can just use regular CREATE

Comment (calbera) [reply cosmos/lib/salt.go]: CREATE2 so that an erc20 with the same name doesn't have the same address

**Pull request closed without merging**